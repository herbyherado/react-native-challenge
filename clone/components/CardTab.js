import React, { Component } from 'react'
import { Text, View, TouchableHighlight, StyleSheet } from 'react-native' 

class CardTab extends Component {
  render() {
    return (
      <TouchableHighlight onPress = { this.props.press }>
        <View style={styles.card}> 
          <Text style={styles.cardText} > { this.props.text }</Text>
        </View>
      </TouchableHighlight>  
    )
  }
};

const styles = StyleSheet.create({
  card: {
    marginTop: 20,
    alignItems: 'center',
    alignSelf: 'center',
    padding: 30,
    width: '95%',
    borderWidth: 1,
    borderBottomColor: 'silver',
    borderTopColor: 'silver'
  },
  cardText: {
    color: '#fff',
    fontSize: 30,
    backgroundColor: 'transparent'
  },
})

export default CardTab