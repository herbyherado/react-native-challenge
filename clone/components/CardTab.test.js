import React, { Component } from 'react'
import enzyme, { shallow, mount } from 'enzyme'
import { Text, View, TouchableHighlight } from 'react-native'
import Adapter from 'enzyme-adapter-react-16'
import CardTab from './CardTab'
import renderer from 'react-test-renderer'

enzyme.configure({ adapter: new Adapter()})

let wrapper

beforeEach(() => {
  wrapper = shallow(<CardTab press={ this.toFighters } text={'Fighters'}/>)
})

describe('<CardTab/>', () => {
  it('should render <CardTab/>', () => {
    expect(wrapper).toBeDefined()
  })
})

describe('<CardTab/> child is rendered', () => {
  it('Should render <TouchableHighlight />', () => {
    expect(wrapper.find(<TouchableHighlight/>)).toBeDefined()
    expect(wrapper.find('TouchableHighlight').length).toBe(1)
    expect(wrapper.find(<TouchableHighlight/>).children()).toBeDefined()
  })
  it('Should render <View/>', () => {
    expect(wrapper.find(<View/>)).toBeDefined()
    expect(wrapper.find('View').length).toBe(1)  
  })
  it('Should render <Text/>', () => {
    expect(wrapper.find(<Text/>)).toBeDefined()
    expect(wrapper.find('Text').length).toBe(1)
  })
})

describe('<CardTab/> props are set', () => {
  it('Should set props in each <CardTab/>', () => {
    expect(wrapper.props()).toBeDefined()
  })
})

describe('<CardTab/> attributes should be rendered', () => {
  it('Should have the expected attributes', () => {
    expect(wrapper.find('Text').at(0).props()).toHaveProperty('style')
    expect(wrapper.find('View').at(0).props()).toHaveProperty('style')
    expect(wrapper.find('TouchableHighlight').at(0).props()).toHaveProperty('onPress')
  })
})

describe('snapshot testing', () => {
  it('snapshot', () => {
    const tree = renderer.create(<CardTab/>)
    expect(tree).toMatchSnapshot()
  })
})
