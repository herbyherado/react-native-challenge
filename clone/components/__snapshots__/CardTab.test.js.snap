// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot testing snapshot 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  hasTVPreferredFocus={undefined}
  hitSlop={undefined}
  isTVSelectable={true}
  nativeID={undefined}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={null}
  testID={undefined}
  tvParallaxProperties={undefined}
>
  <View
    style={
      Object {
        "alignItems": "center",
        "alignSelf": "center",
        "borderBottomColor": "silver",
        "borderTopColor": "silver",
        "borderWidth": 1,
        "marginTop": 20,
        "padding": 30,
        "width": "95%",
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "backgroundColor": "transparent",
          "color": "#fff",
          "fontSize": 30,
        }
      }
    >
       
    </Text>
  </View>
</View>
`;
